include/deal.II/grid/manifold_lib.h

```diff
@@ -127,6 +115,110 @@ class SphericalManifold : public ChartManifold<dim, spacedim, spacedim>
 
 
 /**
+ * Manifold description for a spherical space coordinate system.
+ *
+ * You can use this Manifold object to describe any sphere, circle,
+ * hypersphere or hyperdisc in two or three dimensions, both as a co-dimension
+ * one manifold descriptor or as co-dimension zero manifold descriptor.
+ *
+ * The two template arguments match the meaning of the two template arguments
+ * in Triangulation<dim, spacedim>, however this Manifold can be used to
+ * describe both thin and thick objects, and the behavior is identical when
+ * dim <= spacedim, i.e., the functionality of SphericalManifold<2,3> is
+ * identical to SphericalManifold<3,3>.
+ *
+ * While PolarManifold reflects the usual notion of polar coordinates,
+ * it may not be suitable for domains that contain either the north or
+ * south poles.
+ * Consider for istance the pair of points \f$x_1=(1,\pi/3,0)\f$ and
+ * \f$x_2=(1,\pi/3,\pi)\f$.
+ * These two points would be connented (using a PolarManifold) by the curve
+ * \$[
+ * \begin{align}
+ *   s: [0,1]  & \rightarrow &  \mathbb S^3 \\
+ *           t & \mapsto     &  (1,\pi/3,0) + (0,0,t\pi)
+ * \$]
+ * This curve is not a geodesic on the sphere, and it is not how we
+ * would choose a curve on the sphere. A better one would be the one
+ * passing through the North pole:
+ * \[
+ *  s(t) = x_1 \cos(\alpha(t)) + \kappa \times x_1 \sin(\alpha(t)) +
+ *  \kappa ( \kappa \cdot x_1) (1-\cos(\alpha(t))).
+ * \]
+ * where $\kappa = \frac{x_1 \times \x_2}{\Vert x_1 \times \x_2 \Vert}$
+ * and $\alpha(t) = t * \arccos(x_1 * x_2) $ for $t\in[0,1]$.
+ * Indeed, this is a geodesic, and it is the natural choice when
+ * connecting points on the surface of the sphere.
+ *
+ * This class implements a Manifold that joins any two points in space
+ * by first projecting them on the surface of a sphere with unit
+ * radius, then connecting them with a geodesic, and finally rescaling
+ * the final radius so that the resulting one is the weighted average
+ * of the starting radii. This Manifold is identical to PolarManifold
+ * in dimension two, while for dimension three it returns points that
+ * are more uniformly distributed on the sphere, and it is invariant
+ * with respect to rotations of the coordinate system, therefore
+ * avoiding the problems that PolarManifold has at the poles.
+ *
+ * For mathematical reasons, it is impossible to construct a unique
+ * map of a sphere using only geodesic curves, and therefore, using
+ * this class with MappingManifold is discouraged. If you use this
+ * Manifold to describe the geometry of a sphere, you should use
+ * MappingQ as the underlying mapping, and not MappingManifold.
```

Comment:
I think @ESeNonFossiIo  committed a version of the paper in which this is explained in detail. The main reason is that if you take the average of three points, by pairwise simplification, this process dependent on the order in which you do it, i.e., average of p1, p2, p3 (with weight 1/3) and of p1, p3, p2 (with weight 1/3) do not commute if you construct the lines as geodesics.